ffs/filcold/filcold.go

```diff
@@ -284,12 +291,17 @@ Loop:
 		case <-time.After(timeout):
 			msg := fmt.Sprintf("DealID %d with miner %s tracking timed out after waiting for %.0f hours.", last.DealID, last.Miner, timeout.Hours())
 			fc.l.Log(ctx, msg)
-			return ffs.FilStorage{}, ffs.DealError{ProposalCid: proposal, Message: msg}
+			return ffs.FilStorage{}, ffs.DealError{ProposalCid: proposal, Miner: last.Miner, Message: msg}
 		case di, ok := <-chDi:
 			if !ok {
 				break Loop
 			}
 			last = di
+			select {
+			case dealUpdates <- di:
+			default:
```

Comment:
Hey @jsign maybe I don't know Go switch syntax well, but do I need to put some statement inside the `case dealUpdates <- di:` case? Otherwise, isn't that case getting combined with the `default` case?

Reply:
No, isn't merged. Take a look [here](https://play.golang.org/p/R-JdxLmg-D-). You can commend and uncomment the goroutine to see both situations.

Reply:
(Comment/Uncomment the go routine)